internal/rootcoord/meta_table.go

```diff
@@ -70,6 +70,7 @@ type IMetaTable interface {
 	GetCollectionByName(ctx context.Context, collectionName string, ts Timestamp) (*model.Collection, error)
 	GetCollectionByID(ctx context.Context, collectionID UniqueID, ts Timestamp) (*model.Collection, error)
 	ListCollections(ctx context.Context, ts Timestamp) ([]*model.Collection, error)
+	ListCollectionsFromCache() ([]*model.Collection, error)
```

Comment:
Adding a function in IMetaTable is not a good design.
You can try to implement IMetatable in a new struct named `MetaTableWIthCache` which wraps a `MetaTable` with local cache I think is better.

Reply:
It seems that `ListCollectionsFromCache` can be an internal logic of `ListCollections` and we don't need to check if passed ts is max ts outside.

Reply:
hi~ @jiaoew1991  Thanks for your concern . Adding a new struct will cause a larger change on codes. Because rootcoord compose the interface IMetaTable, and  some ut codes has use the mock of IMetatable. How about Refacting the name of IMetaTable to   IMetaTableWithCache , and then  we make IMetaTableWithCache compose a new interface MemCache.

Reply:
> It seems that `ListCollectionsFromCache` can be an internal logic of `ListCollections` and we don't need to check if passed ts is max ts outside.

hi~ @longjiquan   I find the `restore` method of Core  also call ListCollections to restore from etcd,  which passing the MaxTimestap.  Shall we  modify the usage of ListCollections in `restore`  if we do like you said

Reply:
hi ~ @wg1026688210 , thx for the enhancement. However, anyway, it's correct to check if ts is max inside `ListCollections`.